perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/AttributesManagerMethod.java

```diff
@@ -2653,23 +2657,28 @@ public Void call(ApiCaller ac, Deserializer parms) throws PerunException {
 			for(int i : ids) {
 				attributes.add(ac.getAttributeDefinitionById(i));
 			}
-
 			if (parms.contains("facility")) {
-				if (parms.contains("resource") && parms.contains("group") && parms.contains("user") && parms.contains("member")) {
-					ac.getAttributesManager().removeAttributes(ac.getSession(),
-							ac.getFacilityById(parms.readInt("facility")),
-							ac.getResourceById(parms.readInt("resource")),
-							ac.getGroupById(parms.readInt("group")),
-							ac.getUserById(parms.readInt("user")),
-							ac.getMemberById(parms.readInt("member")),
-							attributes);
-				} else if (parms.contains("resource") && parms.contains("user") && parms.contains("member")) {
-					ac.getAttributesManager().removeAttributes(ac.getSession(),
-							ac.getFacilityById(parms.readInt("facility")),
-							ac.getResourceById(parms.readInt("resource")),
-							ac.getUserById(parms.readInt("user")),
-							ac.getMemberById(parms.readInt("member")),
-							attributes);
+				if (parms.contains("resource")) {
+					if (parms.contains("member")) {
+						if (parms.contains("user")) {
+							if (parms.contains("group")) {
```

Comment:
This is better, but it has problem with else branches of inner if clauses. For this specific use case in my opinion the best way how to do it without lot's of else branches with "missing parameters" exceptions is:
if(parms.contains("resource") && parms.contains("member") && parms.contains("user")) {
___if(parms.contains("group")) {
______call method with group
___} else {
______call method without group
___}
} else {
___throw new RpcException(RpcException.Type.MISSING_VALUE, "text");
}